### PR TITLE
8.7RC umb-block-card absolute path and dont scale SVGs

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Make thumbnails paths work and don't scale SVGs

---
_This item has been added to our backlog [AB#7910](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/7910)_

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/views/components/blockcard/umbBlockCard.component.js [hunk 1]
```diff
@@ -14,9 +14,39 @@
...
+            vm.updateThumbnail();
+
+        }
+        vm.$onDestroy = function () {
+            unwatch();
+        }
+
+        vm.updateThumbnail = function () {
+            if (vm.blockConfigModel.thumbnail == null || vm.blockConfigModel.thumbnail === "") {
+                vm.styleBackgroundImage = "transparent";
```
Comment (bjarnef) [inline src/Umbraco.Web.UI.Client/src/views/components/blockcard/umbBlockCard.component.js:40]: @nielslyngsoe notice `transparent` isn't a valid value for `background-image` but `none` is.

Comment (nielslyngsoe) [reply src/Umbraco.Web.UI.Client/src/views/components/blockcard/umbBlockCard.component.js]: @bjarnef Thanks, what should I have done without you :-D Thanks a lot

**Comment by warrenbuckley** [comment]
Code looks fine - off to test this out with a PNG & SVG

File: src/Umbraco.Web.UI.Client/src/views/components/blockcard/umbBlockCard.component.js [hunk 2]
```diff
@@ -14,9 +14,38 @@
...
+
+        vm.updateThumbnail = function () {
+            if (vm.blockConfigModel.thumbnail == null || vm.blockConfigModel.thumbnail === "") {
+                vm.styleBackgroundImage = "none";
+                return;
+            }
+
+            var path = umbRequestHelper.convertVirtualToAbsolutePath(vm.blockConfigModel.thumbnail);
+            if (path.toLowerCase().endsWith(".svg") === false) {
+                path += "?upscale=false&width=400)";
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/components/blockcard/umbBlockCard.component.js:45]: Remove the closing ) as it generates a URL that is invalid & throws a 500 error
`http://localhost:8700/Media/3bplcy5p/18530280048_459b8b61b2_h.jpg?upscale=false&width=400)`

```suggestion
                path += "?upscale=false&width=400";
```

**Comment by warrenbuckley** [comment]
Should the SVG scale as well, as mine is clipped?

**SVG file contents used**
```
<svg fill="#000000" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64" width="512px" height="512px"><path d="M55.689 39.378l-14.687-6.577c-.289.604-.58 1.206-.75 1.856l14.6 6.538C57.979 42.676 60 45.868 60 49.328V57c0 .551-.449 1-1 1H5c-.551 0-1-.449-1-1v-7.672c0-3.457 2.016-6.646 5.138-8.129l14.252-6.382c-.39-.564-.834-1.087-1.248-1.632l-13.85 6.203C4.469 41.197 2 45.1 2 49.328V57c0 1.654 1.346 3 3 3h54c1.654 0 3-1.346 3-3v-7.672C62 45.1 59.531 41.197 55.689 39.378zM27.92 12.88c2.864 1.39 6.006 2.115 9.19 2.12h2.55c.552 0 1-.448 1-1s-.448-1-1-1h-2.55c-2.883-.004-5.727-.661-8.32-1.92l-1.56-.76c-.905-.442-1.998-.142-2.55.7l-1.6 2.43c-.296.466-.158 1.084.308 1.38.455.289 1.058.165 1.362-.28l1.6-2.43L27.92 12.88z"/><path d="M46.28 18.12c-.141-.036-.285-.056-.43-.06-.172-1.179-.48-2.333-.92-3.44v-.11c.474.057.938.178 1.38.36.51.212 1.095-.03 1.307-.54.079-.19.097-.4.053-.6C46 6.41 40.56 2 33 2h-1c-2.57.011-5.092.701-7.31 2C21 4.14 18.42 6.76 18 10.9c-.185 1.865-.141 3.746.13 5.6.054.331.084.665.09 1 0 .187-.03.373-.09.56-.148.004-.296.024-.44.06-1.65.44-2.34 2.85-1.6 5.61.54 2 1.66 3.5 2.88 4 .623 1.858 1.48 3.628 2.55 5.27l-.003.002c.304.616 1.183 1.678 1.183 1.678C25.16 37.79 28.4 40 32 40c3.584 0 6.811-2.192 9.267-5.271C41.635 34.271 42.48 33 42.48 33c1.059-1.643 1.906-3.414 2.52-5.27 1.23-.46 2.35-2 2.88-4C48.62 20.98 47.88 18.57 46.28 18.12zM45.95 23.25c-.5 1.85-1.44 2.62-1.73 2.62-.459-.018-.872.279-1 .72C41.59 32.26 37.18 38 32 38s-9.59-5.74-11.28-11.4c-.123-.423-.509-.716-.95-.72h-.02c-.29 0-1.24-.78-1.73-2.62-.49-1.84-.05-3.09.18-3.15.099.017.191.058.27.12.468.293 1.085.152 1.378-.316.091-.146.143-.312.152-.484.024-.516.081-1.031.17-1.54.052-.548.025-1.1-.08-1.64-.239-1.707-.269-3.436-.09-5.15C20.31 7.91 22.19 6 25 6c.216 0 .427-.07.6-.2C27.523 4.613 29.74 3.99 32 4h1c6.07 0 10.45 3.1 12.25 8.58-.673-.131-1.37.045-1.9.48-.576.627-.693 1.55-.29 2.3.533 1.294.838 2.672.9 4.07.028.552.498.976 1.049.948.174-.009.344-.063.49-.158.08-.06.172-.101.27-.12C46.01 20.18 46.45 21.4 45.95 23.25zM7 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C8 55.552 7.552 56 7 56zM37 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C38 55.552 37.552 56 37 56zM42 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C43 55.552 42.552 56 42 56zM52 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C53 55.552 52.552 56 52 56zM57 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C58 55.552 57.552 56 57 56zM47 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C48 55.552 47.552 56 47 56zM12 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C13 55.552 12.552 56 12 56zM17 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C18 55.552 17.552 56 17 56zM22 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C23 55.552 22.552 56 22 56zM27 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C28 55.552 27.552 56 27 56zM32 56c-.552 0-1-.448-1-1v-2c0-.552.448-1 1-1s1 .448 1 1v2C33 55.552 32.552 56 32 56z"/></svg>
```

image

**Comment by nielslyngsoe** [comment]
> Should the SVG scale as well, as mine is clipped?

This depends a bit on the content of your thumbnail. If you took a screenshot you probably want it to ´cover´ so you don't get minor gaps here and there. If a designer makes SVG icons, they should make them in the same format or add the desired spacing to their SVG. :-)

**Comment by bjarnef** [comment]
@nielslyngsoe maybe this should work a bit like SVG in `umb-icon` (target for v8.8 I think) where it sanitize the SVG string and insert the inline SVG string. Then it is possible to change color of the SVG icon. Not sure if that is possible at the moment without selecting different SVG files?

**Comment by nielslyngsoe** [comment]
> @nielslyngsoe maybe this should work a bit like SVG in `umb-icon` (target for v8.8 I think) where it sanitize the SVG string and insert the inline SVG string. Then it is possible to change color of the SVG icon. Not sure if that is possible at the moment without selecting different SVG files?

Would be nice. As long as we make sure to still let the developer able to leave the 'icon color' empty and thereby have multiple/original colors in their SVGs. :-)

Feel free to provide a PR if you like :-)

**Comment by warrenbuckley** [comment]
I am merging this in for now

**Comment by bjarnef** [comment]
@nielslyngsoe it also need to ensure images still can be used. Not sure icon picker and color swatches can be used (I guess it allow using custom classes for icons and colors?)
Maybe `umb-icon` could be used (but first available in v8.8) e.g. to use custom SVG or font icon. For images like gif, jpg, png it would still use background image.

**Comment by bjarnef** [comment]
It can use `background-size: 100% 100%`  so make the SVG fit within the box.

image

image

But I also wonder if the SVG should be inline like in `umb-icon` for v8.8
This would make it possible to change e.g. path color in SVG via styling.

**Pull request merged**